tests/infra/test_with_custom_state.py

```diff
@@ -0,0 +1,98 @@
+from eth2spec.test.context import (
+    default_activation_threshold,
+    default_balances,
+    zero_activation_threshold,
+)
+from eth2spec.test.helpers.constants import MINIMAL, PHASE0
+from eth2spec.test.helpers.specs import spec_targets
+from tests.infra.context import _custom_state_cache_dict, with_custom_state
+
+
+class TestWithCustomState:
+    """Test suite for with_custom_state decorator."""
+
+    def setup_method(self):
+        """Clear custom state cache before each test."""
+        _custom_state_cache_dict.clear()
+
+    def teardown_method(self):
+        """Clear custom state cache after each test."""
+        _custom_state_cache_dict.clear()
+
+    def test_with_custom_state_injects_state_view(self):
+        """Test that the decorator injects state and that it differs from default."""
+        spec = spec_targets[MINIMAL][PHASE0]
+
+        def fewer_balances(s):
```

Comment:
we are not testing that this changes the activation_threshold

Reply:
_custom_state_cache_dict is an implementation detail of this decorator and I don't think we should test that. We should test what it actually intents to do, change state balances and activation thresholds